### ISSUE TITLE
unused function parameter

### ISSUE THREAD

**Issue by xxxijustwei** [rmrk-team/evm#213]
This makes my OCD really hard.
plz

```yaml
Warning: Unused function parameter. Remove or comment out the variable name to silence this warning.
  --> @rmrk-team/evm-contracts/contracts/RMRK/extension/RMRKRoyalties.sol:76:9:
   |
76 |         uint256 tokenId,
   |         ^^^^^^^^^^^^^^^
```

**Comment by ThunderDeliverer** [comment]
I'm sorry to hear that.

We strive to have as thoroughly documented source code as possible, which is the result of it. We need to have all function inputs named if we wish to document them.

So I'm sorry to say we can't remove this without impacting the documentation's quality.

**Comment by xxxijustwei** [comment]
Perhaps you can modify it like this, and then it won't prompt unused during compilation:
```solidity
    /**
     * @notice Used to retrieve the information about who shall receive royalties of a sale of the specified token and
     *  how much they will be.
     * @param tokenId ID of the token for which the royalty info is being retrieved
     * @param salePrice Price of the token sale
     * @return receiver The beneficiary receiving royalties of the sale
     * @return royaltyAmount The value of the royalties recieved by the `receiver` from the sale
     */
    function royaltyInfo(
        uint256 tokenId,
        uint256 salePrice
    )
        external
        view
        virtual
        override
        returns (address receiver, uint256 royaltyAmount)
    {
        receiver = _royaltyRecipient;
        royaltyAmount = (salePrice * _royaltyPercentageBps) / 10000;

        tokenId;
    }
```

**Comment by ThunderDeliverer** [comment]
This might remove the warning, but it increases contract size (which is limited) and increases gas consumption (because of the read operation). For these reasons, we can't implement this.

**Comment by steven2308** [comment]
I actually gave it a try yesterday with an idea I had for a while. I duplicated the `IERC2981` definition from OZ into a local file and added the right doc string there. Then, on the implementation I simply used `inheritdoc` and did not set the name for the first parameter.
Surprisingly, the name `tokenId` disappeared from all docs, which is a bigger evil IMO.

**Comment by steven2308** [comment]
I recently tried adding `uint256(tokenId)`, which fixes the issue and does not actually increase size. It is now released on 2.3.2
